templates/IAM/github-oidc-provider.j2

```diff
@@ -0,0 +1,110 @@
+{# Allow Github to access AWS without giving github AWS credentials #}
+{# Must pass in the following templating parameters                 #}
+{#   Repositories:                                                  #}
+{#     - name: "organizations-infra"                                #}
+{#       branch: "master"                                           #}
+
+{# Make template compatible with both sceptre and org-formation     #}
+{% if sceptre_user_data.Repositories is defined %}
```

Comment:
Yes, it will error.  The next line uses `sceptre_user_data.Repositories` so it's required.  Alos the comments at the top of this file specifically says that it's required to pass in `Repositories`.

Reply:
Maybe Jinja is configured differently in org-formation, but with my little test script (see below), I'm seeing the behavior I expected. Specifically, checking if `sceptre_user_data.Repositories` is defined results in an uncaught `UndefinedError`. 

If you indeed tested this exact scenario in org-formation and there wasn't an uncaught exception, then I approve this PR. 

## Using `if sceptre_user_data.Repositories is defined`:

```
Simulating sceptre
Repositories: [{'name': 'organizations-infra', 'branch': 'master'}]

Traceback (most recent call last):
  File "./test.py", line 19, in <module>
    output = template.render(name="org-formation", **repos)
  File "/Users/bgrande/.pyenv/versions/3.8.11/lib/python3.8/site-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/Users/bgrande/.pyenv/versions/3.8.11/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 3, in top-level template code
  File "/Users/bgrande/.pyenv/versions/3.8.11/lib/python3.8/site-packages/jinja2/environment.py", line 474, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'sceptre_user_data' is undefined
```

## Using `if sceptre_user_data is defined`:

```
Simulating sceptre
Repositories: [{'name': 'organizations-infra', 'branch': 'master'}]

Simulating org-formation
Repositories: [{'name': 'organizations-infra', 'branch': 'master'}]
```

## Python Script

```python
from jinja2 import Template

template = Template("""
Simulating {{ name }}
{% if sceptre_user_data.Repositories is defined -%}
    {%- set Repositories = sceptre_user_data.Repositories -%}
{% endif -%}
Repositories: {{ Repositories }}
""")

# Example repositories
repos = {"Repositories": [{"name": "organizations-infra", "branch": "master"}]}

# Simulating sceptre
output = template.render(name="sceptre", sceptre_user_data=repos)
print(output)

# Simulating org-formation
output = template.render(name="org-formation", **repos)
print(output)
```

Reply:
i'm not sure your `# Simulating org-formation` is valid.  org-formation uses [nunjucks](https://mozilla.github.io/nunjucks/) which is different than jinja2.  They both support similar similar language constructs but they are not exactly the same.  

> If you indeed tested this exact scenario in org-formation and there wasn't an uncaught exception

Yes, i have tested running this template with both sceptre and orgf, there is no exception when run with orgf.